_posts/2021-01-11-strategies-for-small-focused-pull-requests.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: epic
+title: "Strategies For Small, Focused Pull Requests"
+date: 2021-01-06
+categories: [tools, github, team, community, engineering]
+author: steve-hicks
+---
+
+A common suggestion for improving pull requests (PRs) is to "make your PR small and focused". I myself gave this suggestion in [a recent article on this very blog about including context in PRs](https://artsy.github.io/blog/2020/08/11/improve-pull-requests-by-including-valuable-context/). 
+
+Like most internet advice, this can feel like the ["draw the rest of the owl"](draw-the-rest-of-the-owl) meme. Even if we're in agreement that I _should_ make a PR smaller...**_how_** do I do it? How do I avoid a big PR when there's a lot of cross-cutting changes to make? How do I create small, focused units of work when I'm building a large feature? How can I overcome my perfectionism and submit a PR that feels incomplete to me because the edges aren't all polished?
+
+<!-- more -->
+
+## What is "small and focused"? 
+
+Not all small PRs are focused. I might sneak five unrelated one-line changes into a PR. While it feels like that will enable me to move quickly, it also runs the risk of four unrelated changes being held up in review because one of them is controversial.
+
+Not all focused PRs are small. I might put an entire feature in one PR, and while it is focused, it's still going to be difficult for you to review the large amount of changes thoroughly. 
+
+To make our PR reviewers' lives easier, we're looking for the intersection of small _and_ focused. Changes that are cohesive and without distractions. Code that accomplishes one thing. TODO: find some examples that show the extremes?
+
```

Comment:
I'm sure you'll find other examples but work I was doing on Purchase last week reminded me of this. [This ticket](https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=41&projectKey=PURCHASE&modal=detail&selectedIssue=PURCHASE-2125) and its substasks (as well as an earlier ticket or two focused on more backend work) comprise one relatively small feature. I was working on [this guy](https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=41&projectKey=PURCHASE&modal=detail&selectedIssue=PURCHASE-2300) and had to catch myself multiple times from implementing the `add` and `edit` parts! My PR for [displaying addresses](https://github.com/artsy/force/pull/6920) doesn't even work on its own, but broken up, it was super manageable to implement!